### PR TITLE
Add support for HEX / UNHEX and ENCODE / DECODE to Presto and Spark

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlglot/dialects/presto.py [hunk 1]
```diff
@@ -106,6 +117,11 @@ def _ts_or_ds_add_sql(self, expression):
     return f"DATE_ADD({unit}, {e}, DATE_PARSE(SUBSTR({this}, 1, 10), {Presto.date_format}))"
 
 
+def _ensure_utf8(charset):
+    if charset.text("this").lower() != "utf-8":
```
Comment (tobymao) [inline sqlglot/dialects/presto.py:121]:
```suggestion
    if charset.name.lower() != "utf-8":
```

File: sqlglot/dialects/presto.py [hunk 2]
```diff
@@ -106,6 +117,11 @@ def _ts_or_ds_add_sql(self, expression):
...
+        raise UnsupportedError(f"Unsupported charset {charset.sql()}")
```
Comment (tobymao) [inline sqlglot/dialects/presto.py:122]:
```suggestion
        raise UnsupportedError(f"Unsupported charset {charset}")
```

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -2623,6 +2631,10 @@ class GroupConcat(Func):
     arg_types = {"this": True, "separator": False}
 
 
+class Hex(Func):
+    arg_types = {"this": True}
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:2635]: We can `pass` here, because `arg_types` is already defined like this in `Expression` (same for `Unhex`).

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/expressions.py (hunk 3 above)
Comment (izeigerman) [reply sqlglot/expressions.py:2635]: good point,thanks

**Pull request merged**